hazelcast/src/test/java/com/hazelcast/client/alto/ClientAltoTest.java

```diff
@@ -338,6 +383,71 @@ public void testAltoDisabledClient_inAltoEnabledCluster() {
         assertEquals("42", map.get("42"));
     }
 
+    @Test
+    public void testAltoClient_heartbeatsToIdleAltoChannels() {
+        Config config = getMemberConfig();
+        Hazelcast.newHazelcastInstance(config);
+
+        ClientConfig clientConfig = getClientConfig();
+        clientConfig.setProperty(HEARTBEAT_INTERVAL.getName(), "1000");
+
+        HazelcastInstance client = HazelcastClient.newHazelcastClient(clientConfig);
+
+        ClientConnectionManager connectionManager = getConnectionManager(client);
+        Collection<ClientConnection> connections = connectionManager.getActiveConnections();
+        assertClientConnectsAllAltoPortsEventually(connections, config.getAltoConfig().getEventloopCount());
+
+        ClientConnection connection = connectionManager.getRandomConnection();
+        assertTrue(connection.isAlive());
+
+        Channel[] altoChannels = connection.getAltoChannels();
+        assertNotNull(altoChannels);
+
+        long now = System.currentTimeMillis();
+        assertTrueEventually(() -> {
+            for (Channel channel : altoChannels) {
+                assertTrue(channel.lastWriteTimeMillis() > now);
+            }
+        });
+    }
+
+    @Test
+    public void testAltoClient_heartbeatsToNotRespondingAltoChannelsTimeouts() {
+        Config config = getMemberConfig();
+        Hazelcast.newHazelcastInstance(config);
+
+        ClientConfig clientConfig = getClientConfig();
+        clientConfig.setProperty(HEARTBEAT_INTERVAL.getName(), "1000");
+        clientConfig.setProperty(HEARTBEAT_TIMEOUT.getName(), "3000");
+
+        HazelcastInstance client = HazelcastClient.newHazelcastClient(clientConfig);
+        ClientConnectionManager connectionManager = getConnectionManager(client);
+        Collection<ClientConnection> connections = connectionManager.getActiveConnections();
+
+        assertClientConnectsAllAltoPortsEventually(connections, config.getAltoConfig().getEventloopCount());
+
+        ClientConnection connection = connections.iterator().next();
+        assertTrue(connection.isAlive());
+
+        // This is a long-running task that will block the Alto thread, and it
```

Comment:
does `map.executeOnKey(1, ...)` work on partition thread or alto thread?

Reply:
One alto thread, which is responsible for possibly many partitions